### ISSUE TITLE
Invalid default value returned when deserializing to nullable string

### ISSUE THREAD

**Issue by vxzms** [prozolic/CsToml#12]
Thank you for creating a high-performance, AOT-compatible TOML library, but I have encountered an issue during testing.

During deserialization, the default value for nullable types is null, except for nullable string which returns string.Empty.

```csharp
[TomlSerializedObject]
public partial class TestClass
{
    [TomlValueOnSerialized] public string? TextNullable {get; set;}
    [TomlValueOnSerialized] public string Text {get; set;}
    [TomlValueOnSerialized] public int? Numberullable {get; set;}
    [TomlValueOnSerialized] public int Number {get; set;}
    [TomlValueOnSerialized] public bool? Isullable {get; set;}
    [TomlValueOnSerialized] public bool Is {get; set;}
    [TomlValueOnSerialized] public DateTime? DateTimeullable {get; set;}
    [TomlValueOnSerialized] public DateTime DateTime {get; set;}
}

var text = ""u8;
var res = CsTomlSerializer.Deserialize<TestClass>(text);
```
Result:
![image](https://github.com/user-attachments/assets/ad80d514-2035-476b-8029-b25ae2d546b8)

**Comment by prozolic** [comment]
Thank you, `string` and `string?`(nullable string) are internally of the same type, so they return the same value, but it is a bug that returns "". 
I'll fix it.

**Comment by prozolic** [comment]
Thank you for your feedback. I have released v1.3.1. Please give it a try.
